docs/DeveloperGuide.md

```diff
@@ -161,9 +161,30 @@ The `TagNoteMap#initTagNoteMapFromNotes()` is exposed in the `Model` interface a
 
 Given below is an example usage scenario and how mapping mechanism behaves at each step.
 
+### Associating Notes and Countries
 
+#### Implementation
+
+The association between `Note` and `Country` is facilitated by `CountryNotesManager`.
+Internally, `CountryNotesManager` stores a list of country notes using an instance of `javafx.collections.ObservableList<CountryNote>`. 
+By storing the list of country notes as an `ObservableList`, the UI would be able to track and immediately reflect any changes to the country notes list.
+
+It implements the following operations:
+* `CountryNotesManager#asUnmodifiableObservableList()` — Returns an unmodifiable `ObservableList<CountryNote>`. 
+* `CountryNotesManager#hasCountryNote(CountryNote countryNote)`  — Returns true if the given `countryNote` already exists in the internal `ObservableList<CountryNote>`.
+* `CountryNotesManager#addCountryNote(CountryNote countryNote)`  — Adds the given `countryNote` to the internal `ObservableList<CountryNote>`.
+* `CountryNotesManager#deleteCountryNote(CountryNote countryNote)`  — Deletes the given `countryNote` from the internal `ObservableList<CountryNote>`.
+
+The following class diagram illustrates how the relevant classes in the `seedu.address.model` package are related to `CountryNote`. 
+
+![Country Class Diagram](images/CountryClass.png)
 
+Given below is a sequence diagram that shows how the `country note add` command works. 
+For brevity, the full command `country note add c/COUNTRY_CODE nt/NOTE_STRING` will be substituted by `country note add`. 
+Note that the `AddressBookParser` parses the user input and returns a `CountryNote` object, which will be passed as an argument to the constructor of `CountryNoteAddCommand`. 
```

Comment:
Slightly inaccurate as `AddressBookParser` returns a `Command` type. Maybe talk about `CountryNoteAddCommandParser` which extracts `Country` and `Note` into `CountryNote`?

Reply:
Yea thanks for catching that mistake, will mention about `CountryNoteAddCommandParser` and also say that it will be excluded from the sequence diagram for brevity.

Added the intermediate call to `Addressbook#addCountryNote(countryNote)`